### PR TITLE
use clusterversions for getting hub cluster id

### PR DESCRIPTION
**Pull request by xrajesh**
**Related Issue:**  open-cluster-management/backlog#<ISSUE_NUMBER>

### Description of changes

/hold
`/hold`

### CODE REVIEW

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

File: pkg/monitor/clustermonitor.go [hunk 1]
```diff
@@ -229,13 +234,29 @@ func (m *Monitor) deleteCluster(managedCluster *clusterv1.ManagedCluster) {
...
+		return "-1"
+	}
+	clusterID, _, err := unstructured.NestedString(versionObj.Object, "spec", "clusterID")
+	if err != nil {
+		glog.V(2).Infof("Failed to get OCP clusterID from version: %v", err)
+		return "-1"
 	}
-	return "-1"
+	lock.Lock()
+	m.ManagedClusterInfo = append(m.ManagedClusterInfo, types.ManagedClusterInfo{
```
Comment (zlayne) [inline pkg/monitor/clustermonitor.go:255]: We only want insights for clusters that are Openshift & version 4.X. Using this resource we can assume it is an Openshift cluster, but we will need to check the version is correct.

Comment (zlayne) [reply pkg/monitor/clustermonitor.go:255]: Looks like we could use `status.desired.version`

Comment (xrajesh) [reply pkg/monitor/clustermonitor.go:255]: status.desired.version - not sure if thats the current version or the one RH. wants you to be in .

**Review by zlayne — APPROVED** [review]

File: pkg/monitor/clustermonitor.go (hunk 1 above)
Comment (zlayne) [reply pkg/monitor/clustermonitor.go:255]: `ACM is not support on hub on openshift 3.x`. Dont need to check version if we dont support less than 4.X. resolving

Comment (jlpadilla) [reply pkg/monitor/clustermonitor.go:255]: Good point. Do we check this for the managed clusters?

Comment (zlayne) [reply pkg/monitor/clustermonitor.go:255]: Yep we check the version for all managed clusters

**Pull request merged**